frontend/src/components/VaultCreationProgress.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div class="flex items-center">
+    <template v-for="(step, index) in steps" :key="step">
+      <div class="flex items-center">
+        <div
+          class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full border-3 cursor-default"
+          :class="{
+            'bg-primary border-primary text-white': index < currentStep,
+            'bg-white border-gray-300 text-black': index > currentStep,
+            'bg-white border-primary text-white': index === currentStep
+          }"
+          :title="tooltipForStep(step)"
+        >
+          <template v-if="index < currentStep">
+            <CheckIcon class="text-white" />
+          </template>
+          <template v-else>
+            <span class="h-1.5 w-1.5 rounded-full block" :class="index === currentStep ? 'bg-primary' : 'bg-gray-300'"/>
```

Comment:
Since this dot should only appear for the current step, it could be larger.

```suggestion
            <span class="h-2.5 w-2.5 rounded-full block" :class="index === currentStep ? 'bg-primary' : 'bg-gray-300'"/>
```